Update README.md (#17)
crossplane/test: README.md

> # test

Reviewer: nitpick: could you wrap all lines at 80 characters :)

> 

Author: Will it be a good idea to include the design of current test workflows? 
Maybe something like this [Prometheus test-infra](https://github.com/prometheus/test-infra/blob/master/prombench/README.md)?
I think it might be too much for this stage, so maybe in the future?

Reviewer: I think we can defer until later on :+1:

## Overview 

> This is the home to automated end-to-end tests for the Crossplane Infrastructure. It contains logic and configuration files for the testing of Crossplane and its stack. 

Reviewer: ```suggestion
This is the home for automated end-to-end tests for the Crossplane ecosystem. It contains logic and configuration files for the testing of Crossplane and its providers. 
```

> It uses Go and Github actions primarily to run tests and automated workflows. To create a new workflow place a workflow file in the `.github` directory of the repository. These workflows install crossplane into a `kind` cluster to run different test scenarios. 

Reviewer: ```suggestion
It uses Go and Github actions primarily to run tests and automated workflows. To create a new workflow place a workflow file in the `.github` directory of the repository.
```

I think we can omit last line since they are not strictly required to execute in that manner.

You can have a look at actions tab to see all the automated workflow runs and their status.

> 

Author: @hasheddan Will it be a good idea to include the current automated workflows? Maybe something like this:
Currently, this repo holds the following tests that run on a daily basis:
- e2e tests against the latest stable release of Crossplane
- provider upgrade tests
- core crossplane upgrade test

Reviewer: That sounds good, maybe link to the workflow file for each of them.

### Repo Structure:

> - `.github` - contains all the scheduled Github workflows for testing crossplane infrastructure. 

Reviewer: ```suggestion
- `.github` - contains all the scheduled Github workflows for testing crossplane components. 
```

- `apis` - contains all the API types useful for running Go tests.
- `config` - contains YAML files that use the above mentioned API types.
- `test` - contains all the test scripts with subdirectory name denoting the component under consideration. 

## Background and Purpose

> As Crossplane and its providers have grown and evolved, the surface area for potential bugs has increased as well. Unit tests and automated integration tests at stack level are effective for determining if the stack can successfully be installed into a Crossplane Kubernetes cluster, it does not go so far as to test any of the controllers beyond that they start. This raised the need for an automated end-to-end test infrastructure. 

Reviewer: ```suggestion
As Crossplane and its providers have grown and evolved, the surface area for potential bugs has increased as well. Unit tests and automated integration tests at the provider level are effective for determining if the provider can successfully be installed into a Crossplane Kubernetes cluster, but they do not go so far as to test any of the controllers beyond that they start. This raised the need for an automated end-to-end test infrastructure. 
```

> The primary objective of this repo is to ensure stable behaviour across crossplane and to verify how it behaves with other parts of its stack. It is also important to ensure that upgrades of Crossplane and its providers from stable to master should also not broken. 

Reviewer: ```suggestion
The primary objective of this repo is to ensure stable behaviour across crossplane and to verify how it behaves with other parts of its stack. It is also important to ensure that upgrades of Crossplane and its providers from stable versions to latest builds are not broken. 
```
